internals/src/Tizen.Peripheral/Tizen.Peripheral/Uart.cs

```diff
@@ -0,0 +1,354 @@
+    }
+
+    /// <summary>
+    /// Enumeration for hardware flow control.
+    /// </summary>
+    /// <since_tizen> 8 </since_tizen>
+    public enum HardwareFlowControl
+    {
+        /// <summary>No hardware flow control.</summary>
+        None = 0,
+
+        /// <summary>Automatic RTS/CTS hardware flow control.</summary>
+        AutoRtsCts
+    }
+
+    /// <summary>
+    /// Enumeration for software flow control.
+    /// </summary>
+    /// <since_tizen> 8 </since_tizen>
+    public enum SoftwareFlowControl
+    {
+        /// <summary>No software flow control.</summary>
+        None = 0,
+
+        /// <summary>XON/XOFF software flow control.</summary>
+        XonXoff,
+    }
+
+    /// <summary>
+    /// The class allows applications to communicate via UART platform's bus.
+    /// </summary>
+    /// <privilege>http://tizen.org/privilege/peripheralio</privilege>
+    /// <since_tizen> 8 </since_tizen>
+    public class Uart : IDisposable
+    {
+        private BaudRate _baudRate;
+        private ByteSize _byteSize;
+        private Parity _parity;
+        private StopBits _stopBits;
+        private HardwareFlowControl _hardwareFlowControl;
+        private SoftwareFlowControl _softwareFlowControl;
+        private bool _disposed = false;
+
+        /// <summary>
+        /// Native handle to I2c.
+        /// </summary>
+        private IntPtr _handle;
+
+        /// <summary>
+        /// Opens the UART slave device.
+        /// </summary>
+        /// <param name="port">The UART port number that the slave device is connected.</param>
+        /// <since_tizen> 8 </since_tizen>
+        public Uart(int port)
+        {
+            var ret = NativeUart.Open(port, out IntPtr handle);
+            if (ret != Internals.Errors.ErrorCode.None)
+                throw ExceptionFactory.CreateException(ret);
+
+            _handle = handle;
+        }
+
+        /// <summary>
+        /// Closes the UART slave device.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        public void Close() => Dispose();
+
+        /// <summary>
+        /// Disposes Uart object.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        public void Dispose() => Dispose(true);
+
+        /// <summary>
+        /// Disposes Uart object.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (_disposed)
+            {
+                return;
+            }
+
+            if (disposing)
+            {
+                NativeUart.Close(_handle);
+            }
+
+            _disposed = true;
+        }
+
+        /// <summary>
+        /// Reads the bytes data from the UART slave device.
+        /// </summary>
+        /// <param name="dataOut">The output byte array.</param>
+        /// <returns>The array of bytes.</returns>
+        /// <since_tizen> 8 </since_tizen>
+        public void Read(byte[] dataOut)
```

Comment:
How about standard style?
```suggestion
        public int Read(byte[] buffer, int offset, int count);
```

Reply:
I followed Microsoft [API](https://docs.microsoft.com/en-us/uwp/api/windows.devices.spi.spidevice.write?view=winrt-19041#Windows_Devices_Spi_SpiDevice_Write_System_Byte___), it only has 
```csharp
public void Write(byte[])
```
signature

Reply:
Did you see, MS API?
``` c#
public int Read (byte[] buffer, int offset, int count);
```

https://docs.microsoft.com/ko-kr/dotnet/api/system.io.ports.serialport.read?view=dotnet-plat-ext-3.1#System_IO_Ports_SerialPort_Read_System_Byte___System_Int32_System_Int32_